docs/docs/topic-guides/custom-job-types.md

```diff
@@ -0,0 +1,90 @@
+
+
+## Requester node
+
+When a job is received at the requester node, it determines how to process the job based
+on the engine type given in the job specification.  For engine types that are docker or 
+webassembly, they are processed immediately.  
+
+Other engine types are processed by the [translation package](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go) and tasks within the job translated by the server-side job translators, where one is provided for each custom job type.  These [translators](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go#L34C1-L38C2) are responsible for updating the task so that it is able to run on either the docker or webassembly runtimes.  
+
+Once a job has been translated, for example: Python -> Docker then the new job ID is returned to the client as if it were a normal job. The newly translated job may have a dependency on an external container which performs extra-steps during deployment, it is up to the translator to ensure it specifies the correct container or wasm module.
+
+
+## Initial job types 
+
+The initial implementation of custom job types includes job types for Python and DuckDB. Both of these custom job types support the `--code` parameter to simplify the code being sent to  the requester node - for python this is likely python code, for duckdb it is expected to be sql.  Command line parameters that follow the job type are used as the parameters to the translator, so `python /code/app.py /inputs/data.txt` will result in parameters that are `['python', '/code/app.py', '/inputs/data.txt']`.  Any flags that need to be passed to the python process, for instance should app.py require -f for the final parameter, must be sent after the bash -- separator.  
+
+### Dependencies
+
+For Python, there is a likelihood that the code to be executed has dependencies, and these can be specified one of several ways.  The docker image for the Python executor has a specific entrypoint that attempts to determine and then install the requirements.
+
+In a single file attachment (via --code) any pip commands in the module documentation are executed before the code, for instance the following code will install colorama before interpreting the script:
+
+```python
+"""
+pip install colorama 
+"""
+
+from colorama import Fore 
+```
+
+If a `pyproject.toml` is found in a directory specified by `--code`, then it is used to install the dependencies.  If the file has a poetry section, then `poetry install` is used instead.
+
+A `requirements.txt` file will be used with `pip install -r` to install requirements, if found in the specified directory.  
```

Comment:
Does this imply the compute nodes running the job need to have networking enabled? And/or that the client submitting the job most request a networking job?